core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreReaderThread.java

```diff
@@ -147,7 +131,7 @@ public void doWork() {
           V message = null;
           try {
             message =
-                messageBytes == null ? null : serializer.deserializeValue(messageKey, messageBytes);
+                record.value() == null ? null : serializer.deserializeValue(messageKey, record.value());
           } catch (SerializationException e) {
             log.error("Failed to deserialize a schema or config update", e);
```

Comment:
I'll throw an exception here. I'll do the same for the other occurrence you commented on, and an additional occurrence (deserializing the key). Great suggestion.